### PR TITLE
[core] Extend model serialize API to support std::filesystem::path

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Extend `Serialize` and `save_model` API to support cpp17 path class

### Tickets:
 - CVS-158889

### CODE REVIEW

File: src/core/include/openvino/core/graph_util.hpp [hunk 1]
```diff
@@ -288,27 +292,46 @@ bool replace_node_update_name(const std::shared_ptr<Node>& target, const std::sh
...
+void serialize(const std::shared_ptr<const ov::Model>& m,
+               const Path& xml_path,
+               const Path& bin_path = {""},
+               ov::pass::Serialize::Version version = ov::pass::Serialize::Version::UNSPECIFIED) {
+    serialize(m, xml_path.string(), bin_path.string(), version);
+}
+#endif
+/// \}
+
+// TODO CPP17
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/graph_util.hpp:313]: is it valid TODO?

**Review by ilya-lavrenov — APPROVED** [review]

**Review by olpipi — COMMENTED** [review]
Is this feature for python api? Or is there any other specific use case?

**Review by olpipi — APPROVED** [review]

File: src/core/include/openvino/core/graph_util.hpp (hunk 1 above)
Comment (praasz) [reply src/core/include/openvino/core/graph_util.hpp]: No, removed.

**Review by barnasm1 — APPROVED** [review]

**Comment by praasz** [comment]
> Is this feature for python api? Or is there any other specific use case?

It is more for cpp API when OV is used in project which use cpp17,  there can be some compilation issue if implicit cast to string from path will not work.

**Review by t-jankowski — APPROVED** [review]

**Pull request merged**